src/libraries/System.Runtime.Serialization.BinaryFormat/src/System/Runtime/Serialization/BinaryFormat/BinaryArrayType.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Runtime.Serialization.BinaryFormat;
+
+/// <summary>
+/// Binary array type.
+/// </summary>
+/// <remarks>
+/// BinaryArrayType enumeration is described in <see href="https://learn.microsoft.com/openspecs/windows_protocols/ms-nrbf/4dbbf3a8-6bc4-4dfc-aa7e-36a35be6ff58">[MS-NRBF] 2.4.1.1</see>.
+/// </remarks>
+#if SYSTEM_RUNTIME_SERIALIZATION_BINARYFORMAT
+public
+#else
+internal
+#endif
+enum BinaryArrayType : byte
+{
+    /// <summary>
+    ///  A single-dimensional array.
+    /// </summary>
+    Single = 0,
+
+    /// <summary>
+    ///  An array whose elements are arrays. The elements of a jagged array can be of different dimensions and sizes.
+    /// </summary>
+    Jagged = 1,
+
+    /// <summary>
+    ///  A multi-dimensional rectangular array.
+    /// </summary>
+    Rectangular = 2,
+
+    /// <summary>
+    ///  A single-dimensional array where the lower bound index is greater than 0.
```

Comment:
I know that "greater than 0" is actually from [MS-NRBF], but the code (and the spec) seem to allow 0.  Especially for Jagged and Rectangular.

Reply:
I am not sure if we should even support the offset arrays. Not to mention arrays that pretend to be of a custom offset, but using zero offsets only.

@JeremyKuhne @GrabYourPitchforks thoughts?

Reply:
I don't think we should as it is highly unlikely that someone is dependent on this as you can't express such a concept in C# or VB, you have to deliberately create such arrays with the `Array` class. They're effectively unsafe as no normal written code expects arrays to have a length of 1 and have `[0]` be invalid.